fw/ss_skb.c

```diff
@@ -1051,11 +1054,14 @@ ss_skb_cutoff_data(struct sk_buff *skb_head, TfwStr *str, int skip, int tail)
 		}
 
 		skb = c->skb;
-		if (skb->next != skb_head) {
+		is_single = (skb == skb_head && skb->next == skb_head);
+
+		if (skb->next) {
```

Comment:
Isn't this a circular list (tail pointing to the head)? IIRC we deal with a NULL terminated and circular skb lists...